dolphin/tests/unit/drivers/dell_emc/vmax/test_alert_handler.py

```diff
@@ -0,0 +1,86 @@
+# limitations under the License.
+
+import unittest
+
+import pytest
+from oslo_utils import importutils
+
+from dolphin import exception
+
+
+class AlertHandlerTestCase(unittest.TestCase):
+    ALERT_HANDLER_CLASS = 'dolphin.drivers.dell_emc.vmax.alert_handler' \
+                          '.AlertHandler'
+
+    def _get_alert_handler(self):
+        alert_handler_class = importutils.import_class(
+            self.ALERT_HANDLER_CLASS)
+        alert_handler = alert_handler_class()
+        return alert_handler
+
+    def _get_fake_alert_info(self):
+        alert_info = {'storage_id': 'abcd-1234-56789',
+                      'storage_name': 'Dell EMC', 'vendor': 'Dell EMC',
+                      'model': 'vmax', 'connUnitEventId': 79,
+                      'connUnitName': '000192601409',
+                      'connUnitEventType': 'topology',
+                      'connUnitEventDescr': 'Symmetrix 000192601409 FastSRP '
+                                            'SRP_1 : Remote (SRDF) diagnostic '
+                                            'event trace triggered.',
+                      'connUnitEventSeverity': 'warning',
+                      'connUnitType': 'storage-subsystem',
+                      'emcAsyncEventSource': 'symmetrix',
+                      'emcAsyncEventCode': '1050',
+                      'emcAsyncEventComponentType': '1051',
+                      'emcAsyncEventComponentName': 'SRP_1'}
+
+        return alert_info
+
+    def test_parse_alert_with_all_necessary_info(self):
+        """ Success flow with all necessary parameters"""
+        alert_handler_inst = self._get_alert_handler()
+        alert = self._get_fake_alert_info()
+        expected_alert_model = {'me_dn': alert['storage_id'],
+                                'me_name': alert['storage_name'],
+                                'manufacturer': alert['vendor'],
+                                'location': 'Component type: Symmetrix Disk '
+                                            'Group,Component name: SRP_1',
+                                'event_type': alert['connUnitEventType'],
+                                'severity': alert['connUnitEventSeverity'],
+                                'probable_cause': alert['connUnitEventDescr'],
+                                'me_category': alert['connUnitType'],
+                                'native_me_dn': alert['connUnitName'],
+                                'alarm_id': alert['emcAsyncEventCode'],
+                                'alarm_name':
+                                    'SYMAPI_AEVENT2_UID_MOD_DIAG_TRACE_TRIG'
+                                }
+        context = {}
+        alert_model = alert_handler_inst.parse_alert(context, alert)
+
+        # Not comparing dicts as it has some varying para like occur_time
+        # Verify that relevant fields are matching with expected model fields
+        for key in expected_alert_model:
```

Comment:
can be combined to one call assertDictEqual ? possible?

Reply:
Thought about it. As mentioned in comment, there is current time filled in function which will make dict comparison fail

Reply:
if you use dolphin.test.TestCase, you can use assertDictEqual.

Reply:
Done